pysollib/games/hitormiss.py

```diff
@@ -0,0 +1,162 @@
+        VARIABLE_REDEALS
+
+
+# ************************************************************************
+# *
+# ************************************************************************
+
+class HitOrMiss_Foundation(AbstractFoundationStack):
+    def acceptsCards(self, from_stack, cards):
+        game = self.game
+        return game.rank == cards[0].rank  # check the rank
+
+    def getHelp(self):
+        return _('Foundation. Place cards here that match the declared rank.')
+
+
+class HitOrMiss_Talon(WasteTalonStack):
+    def canDealCards(self):
+        game = self.game
+        if (game.deadDeals == 2 and len(self.cards) == 0 and
+                self.max_rounds == VARIABLE_REDEALS):
+            return False
+        return True
+
+    def dealCards(self, sound=False):
+        game = self.game
+        old_state = game.enterState(game.S_FILL)
+        game.saveStateMove(2 | 16)  # for undo
+        if len(self.cards) > 0:
+            game.rank = game.rank + 1
+        else:
+            game.deadDeals = game.deadDeals + 1
+        if game.rank > 12:
+            game.rank = 0
+        game.saveStateMove(1 | 16)  # for redo
+        game.leaveState(old_state)
+        return super(HitOrMiss_Talon, self).dealCards(sound)
+
+    def getHelp(self):
+        return _('Talon. Unlimited redeals, until running through the deck '
+                 'twice with no hits.')
+
+
+class HitOrMiss_Waste(WasteStack):
+    def moveMove(self, ncards, to_stack, frames=-1, shadow=-1):
+        game = self.game
+        if to_stack in game.s.foundations:
+            old_state = game.enterState(game.S_FILL)
+            game.saveStateMove(2 | 16)  # for undo
+            game.deadDeals = 0
+            game.saveStateMove(1 | 16)  # for redo
+            game.leaveState(old_state)
+        super(HitOrMiss_Waste, self).moveMove(ncards, to_stack, frames, shadow)
+        game.s.talon.dealCards()
+
+
+# ************************************************************************
+# * HitOrMiss
+# ************************************************************************
+
+class HitOrMiss(Game):
+
+    def createGame(self):
+        layout, s = Layout(self), self.s
+        self.rank = -1
+        self.deadDeals = 1
+        self.max_rounds = VARIABLE_REDEALS
+        self.setSize(layout.XM+4*layout.XS, layout.YM+2*layout.YS)
+        x, y = layout.XM+3*layout.XS//2, layout.YM
+        stack = HitOrMiss_Foundation(x, y, self, ANY_SUIT,
+                                     dir=0, mod=13, max_move=0, max_cards=52)
+        s.foundations.append(stack)
+        layout.createText(stack, 'ne')
+        x, y = layout.XM+layout.XS, layout.YM+layout.YS
+        s.talon = HitOrMiss_Talon(x, y, self,
+                                  max_rounds=VARIABLE_REDEALS, num_deal=1)
+        layout.createText(s.talon, 'nw')
+
+        x += layout.XS
+        s.waste = HitOrMiss_Waste(x, y, self)
+        layout.createText(s.waste, 'ne')
+        x += layout.XS*1.1
```

Comment:
delimit the asterisk with spaces `x += layout.XS * 1.1`. also applies to other binary infix operators

Reply:
Done.